### PR TITLE
Bug 1802262:  Use polymorphic helpers from exec, when reading a pod 

### PR DESCRIPTION
**Pull request by soltysh**
/assign @sallyom @smarterclayton 

4.3 backport of https://github.com/openshift/oc/pull/260

### CODE REVIEW

**Comment by bparees** [comment]
skipping this one for approval this week due to bug severity and the number of other changes already accepted.

File: pkg/cli/rsh/rsh.go [hunk 1]
```diff
@@ -105,27 +93,35 @@ func NewCmdRsh(name string, parent string, f kcmdutil.Factory, streams genericcl
...
-	cmd.Flags().BoolVarP(&options.DisableTTY, "no-tty", "T", false, "Disable pseudo-terminal allocation")
-	cmd.Flags().StringVar(&options.Executable, "shell", DefaultShell, "Path to the shell command")
-	cmd.Flags().IntVar(&options.Timeout, "timeout", 10, "Request timeout for obtaining a pod from the server; defaults to 10 seconds")
-	cmd.Flags().StringVarP(&options.ContainerName, "container", "c", "", "Container name; defaults to first container")
+	kcmdutil.AddPodRunningTimeoutFlag(cmd, defaultPodRshTimeout)
+	cmd.Flags().BoolVarP(&options.ForceTTY, "tty", "t", options.ForceTTY, "Force a pseudo-terminal to be allocated")
+	cmd.Flags().BoolVarP(&options.DisableTTY, "no-tty", "T", options.DisableTTY, "Disable pseudo-terminal allocation")
+	cmd.Flags().StringVar(&options.Executable, "shell", options.Executable, "Path to the shell command")
+	cmd.Flags().IntVar(&options.Timeout, "timeout", options.Timeout, "Request timeout for obtaining a pod from the server; defaults to 10 seconds")
+	cmd.Flags().MarkDeprecated("timeout", "use --request-timeout, instead.")
```
Comment (smarterclayton) [inline pkg/cli/rsh/rsh.go]: Backporting a deprecation doesn't change the timeline (note)

**Comment by eparis** [comment]
why backport this at all? oc rsh to a deployment works just fine in 4.3, right?

File: pkg/cli/rsh/rsh.go (hunk 1 above)
Comment (soltysh) [reply pkg/cli/rsh/rsh.go]: My bad, this should not be here.

**Comment by soltysh** [comment]
> why backport this at all? oc rsh to a deployment works just fine in 4.3, right?

Deployment - yes, but DaemonSet - nope. This makes the code more resilient. Previous option was very manual.

**Comment by soltysh** [comment]
Rebased, but this needs https://github.com/openshift/origin/pull/24660 to land.

**Pull request merged**